### ISSUE TITLE
Tests don't pass using PHP 7.4

### ISSUE THREAD

**Issue by eclipxe13** [genkgo/xsl#26]
Hi, I think the library is not compatible with PHP 7.4. It does not work on my machine (tm).

Travis-CI is building PHP `7.4snapshot` but that is not the same as current PHP 7.4.
Just replace on `.travis` `s/7.4snapshot/7.4/g`

I check using a docker `debian:bullseye` image to avoid a problem with my machine and it also fail. Didn't use docker `php:7.4` image because extensions `xsl` and `intl` are not installed by default.

Steps I did to reproduce the problem

```text
docker run -it debian:bullseye /bin/bash
apt update
apt install git zip composer php-xsl php-intl
git clone https://github.com/genkgo/xsl.git genkgo-xsl
cd genkgo-xsl
composer install
vendor/bin/phpunit -c phpunit.xml
```

PHPUnit output:

```text
PHPUnit 7.5.20 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.4.5
Configuration: /home/eclipxe/work/github/eclipxe/genkgo-xsl/phpunit.xml

.....................................................EEE.......  63 / 217 ( 29%)
............................................................... 126 / 217 ( 58%)
............................................................... 189 / 217 ( 87%)
............................                                    217 / 217 (100%)

Time: 139 ms, Memory: 8.00 MB

There were 3 errors:

1) Genkgo\Xsl\Unit\XsltProcessorTest::testVersion1
Genkgo\Xsl\Exception\TransformationException: Transformation failed: Transformation failed: XSLTProcessor::importStylesheet(): compilation error: file gxsl://localhost/home/eclipxe/work/github/eclipxe/genkgo-xsl/test/Stubs/version-1.0.xsl#root line 4 element template

/home/eclipxe/work/github/eclipxe/genkgo-xsl/src/XsltProcessor.php:99
/home/eclipxe/work/github/eclipxe/genkgo-xsl/test/Unit/XsltProcessorTest.php:46

Caused by
Genkgo\Xsl\Exception\TransformationException: Transformation failed: XSLTProcessor::importStylesheet(): compilation error: file gxsl://localhost/home/eclipxe/work/github/eclipxe/genkgo-xsl/test/Stubs/version-1.0.xsl#root line 4 element template

/home/eclipxe/work/github/eclipxe/genkgo-xsl/src/XsltProcessor.php:82
/home/eclipxe/work/github/eclipxe/genkgo-xsl/src/XsltProcessor.php:90
/home/eclipxe/work/github/eclipxe/genkgo-xsl/test/Unit/XsltProcessorTest.php:46

2) Genkgo\Xsl\Unit\XsltProcessorTest::testVersion2
Genkgo\Xsl\Exception\TransformationException: Transformation failed: Transformation failed: XSLTProcessor::importStylesheet(): compilation error: file gxsl://localhost/home/eclipxe/work/github/eclipxe/genkgo-xsl/test/Stubs/version-2.0.xsl#root line 4 element template

/home/eclipxe/work/github/eclipxe/genkgo-xsl/src/XsltProcessor.php:99
/home/eclipxe/work/github/eclipxe/genkgo-xsl/test/Unit/XsltProcessorTest.php:60

Caused by
Genkgo\Xsl\Exception\TransformationException: Transformation failed: XSLTProcessor::importStylesheet(): compilation error: file gxsl://localhost/home/eclipxe/work/github/eclipxe/genkgo-xsl/test/Stubs/version-2.0.xsl#root line 4 element template

/home/eclipxe/work/github/eclipxe/genkgo-xsl/src/XsltProcessor.php:82
/home/eclipxe/work/github/eclipxe/genkgo-xsl/src/XsltProcessor.php:90
/home/eclipxe/work/github/eclipxe/genkgo-xsl/test/Unit/XsltProcessorTest.php:60

3) Genkgo\Xsl\Unit\XsltProcessorTest::testVersion3
Genkgo\Xsl\Exception\TransformationException: Transformation failed: Transformation failed: XSLTProcessor::importStylesheet(): compilation error: file gxsl://localhost/home/eclipxe/work/github/eclipxe/genkgo-xsl/test/Stubs/version-3.0.xsl#root line 4 element template

/home/eclipxe/work/github/eclipxe/genkgo-xsl/src/XsltProcessor.php:99
/home/eclipxe/work/github/eclipxe/genkgo-xsl/test/Unit/XsltProcessorTest.php:74

Caused by
Genkgo\Xsl\Exception\TransformationException: Transformation failed: XSLTProcessor::importStylesheet(): compilation error: file gxsl://localhost/home/eclipxe/work/github/eclipxe/genkgo-xsl/test/Stubs/version-3.0.xsl#root line 4 element template

/home/eclipxe/work/github/eclipxe/genkgo-xsl/src/XsltProcessor.php:82
/home/eclipxe/work/github/eclipxe/genkgo-xsl/src/XsltProcessor.php:90
/home/eclipxe/work/github/eclipxe/genkgo-xsl/test/Unit/XsltProcessorTest.php:74

ERRORS!
Tests: 217, Assertions: 315, Errors: 3.
```

I don't have the underlying understanding about how this library works, so I don't really know why is this happening.
As far I can see, the problem is related to the object returned by `XsltProcessor::getTranspiledStyleSheet()` wich is created without any problem and the error occurs when try to send it to `parent::importStylesheet`.

Thanks for your great work!

**Comment by eclipxe13** [comment]
As @peter279k point out, Travis pass on building using PHP 7.4 according to his PR #25.

It builds, I can't figure it out why.

Just try to run tests using docker php:7.4 and fail, docker debian:buster and also fail.
Could it might be related to `libxslt1.1` OS version?

I also found that `test/Stubs/version-{1,2,3}.0.xsl` (the only failing tests uses those files) contains templates without `name` nor `match`, according to MDN `match` is mandatory if `name` does not exists: <https://developer.mozilla.org/en-US/docs/Web/XSLT/Element/template>.
Add `name="foo"` to template element and the tests pass.

**Comment by peter279k** [comment]
@eclipxe13, thanks for your reply.

After reproducing your issue, I've found where the problem is.

The problem is about the `libxslt1.1` system package version.

If I use the `ubuntu:18.04` on my Docker, it will be worked successfully with `php-7.4` and `libxslt1.1` `1.1.29-5` version.

After using your `debian:bullseye` or `ubuntu:20.04`, it will be failed to run with `php-7.4` and `libxslt1.1` `1.1.34-4` version.

When I downgrade `libxslt1.1` to `1.1.29-5` version on `ubuntu:20.04`, it will be successful again.

**Comment by eclipxe13** [comment]
I follow the same path and found something similar:

`grep -i -P "(libxsl|libxml)"`

On Travis: <https://travis-ci.com/github/eclipxe13/xsl/builds/176553910>

```
libxml Version => 2.9.3
libxml
libXML support => active
libXML Compiled Version => 2.9.3
libXML Loaded Version => 20903
libXML streams => enabled
libxml2 Version => 2.9.3
libxslt Version => 1.1.28
libxslt compiled against libxml Version => 2.9.3
```
On debian:bullseye

```
libxml Version => 2.9.10
libxml
libXML support => active
libXML Compiled Version => 2.9.10
libXML Loaded Version => 20910
libXML streams => enabled
libxml2 Version => 2.9.10
libxslt Version => 1.1.34
libxslt compiled against libxml Version => 2.9.10
```

**Comment by eclipxe13** [comment]
@peter279k Me I suggest modify the files `test/Stubs/version-{1,2,3}.0.xsl` changing `<xsl:template>` to `<xsl:template name="foo">` and include it on your PR ?

This would prevent a future issue when Travis-CI upgrades it's libraries or trying to contribute using PHP 7.4.

**Comment by peter279k** [comment]
Sure. I'll modify this file you mention on my PR :).

**Comment by frederikbosch** [comment]
Fixed by #25
